exe-unit/src/service/transfer.rs

```diff
@@ -269,73 +240,66 @@ impl Handler<DeployImage> for TransferService {
-        let final_path = self.cache.to_final_path(&cache_name);
+        let src_url = actor_try!(TransferUrl::parse_with_hash(image, "file"));
+        let src_name = actor_try!(Cache::name(&src_url));
+        let path = self.cache.to_final_path(&src_name).to_path_buf();
 
-        log::info!(
-            "Deploying from {:?} to {:?}",
-            source_url.url,
-            final_path.to_path_buf()
-        );
+        log::info!("Deploying from {:?} to {:?}", src_url.url, path);
 
-        let args = TransferArgs::default();
         #[cfg(not(feature = "sgx"))]
         {
-            let from_provider = actor_try!(self.provider(&source_url));
-            let to_provider: FileTransferProvider = Default::default();
-            let temp_path = self.cache.to_temp_path(&cache_name);
-            let temp_url = Url::from_file_path(temp_path.to_path_buf()).unwrap();
+            let path_tmp = self.cache.to_temp_path(&src_name).to_path_buf();
 
-            let address = ctx.address();
-            let (abort, reg) = Abort::new_pair();
+            let src = actor_try!(self.provider(&src_url));
+            let dst: Rc<FileTransferProvider> = Default::default();
+            let dst_url = TransferUrl {
+                url: Url::from_file_path(&path_tmp).unwrap(),
+                hash: None,
+            };
 
+            let handles = self.abort_handles.clone();
             let fut = async move {
-                let final_path = final_path.to_path_buf();
-                let temp_path = temp_path.to_path_buf();
-
-                let stream_fn = || Self::source(from_provider.clone(), &source_url, &args);
-                let sink_fn = || to_provider.destination(&temp_url, &args);
-
-                if final_path.exists() {
-                    log::info!("Deploying cached image: {:?}", final_path);
-                    return Ok(Some(final_path));
+                if path.exists() {
+                    log::info!("Deploying cached image: {:?}", path);
+                    return Ok(Some(path));
                 }
 
+                let (abort, reg) = Abort::new_pair();
                 {
-                    let _guard = AbortHandleGuard::register(address, abort).await?;
-                    let retry = retry_transfer(stream_fn, sink_fn, Retry::default());
+                    let ctx = Default::default();
+                    let retry = transfer_with(src, &src_url, dst, &dst_url, &ctx);
+
+                    let _guard = AbortHandleGuard::register(handles, abort);
                     Ok(Abortable::new(retry, reg)
                         .await
                         .map_err(TransferError::from)??)
                 }
                 .map_err(|e: Error| {
-                    log::warn!("Removing temporary download file: {}", temp_path.display());
-                    let _ = std::fs::remove_file(&temp_path);
+                    log::warn!("Removing temporary download file: {}", path_tmp.display());
```

Comment:
pls do not remove after failue, as next invocation can utilise already existing part of the file being downloaded